x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/entsearch/engine/EngineIndexService.java

```diff
@@ -285,14 +292,23 @@ private void updateEngine(Engine engine, ActionListener<IndexResponse> listener)
      *
      */
     public void deleteEngine(String engineName, ActionListener<DeleteResponse> listener) {
+
         try {
             // TODO Delete alias when Engine is deleted
             final DeleteRequest deleteRequest = new DeleteRequest(ENGINE_ALIAS_NAME).id(engineName)
                 .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE);
-            clientWithOrigin.delete(deleteRequest, listener);
+
+            clientWithOrigin.delete(deleteRequest, listener.delegateFailure((delegate, deleteResponse) -> {
```

Comment:
Is there a good example of how to chain a delegated failure & delegated response together? I'd like to chain the alias removal on a successful deletion response.

Reply:
I did something similar in the `putEngine` method. AFAIK there's no construct for delegating both failure and success, and had to implement both `onResponse` and `onFailure` in a new `ActionListener`.

Reply:
This requires some fiddling. Will create a new issue to support alias management.